src/proxy.rs

```diff
@@ -50,7 +51,7 @@ impl Proxy {
     pub async fn new(
         cfg: config::Config,
         workloads: WorkloadInformation,
-        cert_manager: Box<dyn CertificateProvider>,
+        cert_manager: Arc<Mutex<Box<dyn CertificateProvider>>>,
```

Comment:
I don't think we need Box if its already in an arc mutex

Reply:
But I am not sure why we need the mutex anyways, the CertificateProvider already has one as part of it.

Maybe https://github.com/istio/ztunnel/pull/325 resolves this?

Reply:
would just an `Arc<dyn CertificateProvider>` suffice then?

Reply:
You may just want to call .clone() on it instead? The CertificateProviders internally use Arc already so they are cheap to clone

Reply:
makes sense